homeassistant/components/sensor/gearbest.py

```diff
@@ -0,0 +1,134 @@
+"""
+Parse prices of a item from gearbest.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.gearbest/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_interval
+from homeassistant.const import (CONF_NAME, CONF_ID, CONF_URL, CONF_CURRENCY)
+
+REQUIREMENTS = ['gearbest_parser==1.0.5']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ITEMS = 'items'
+
+ICON = 'mdi:coin'
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=2*60*60)  # 2h
+MIN_TIME_BETWEEN_CURRENCY_UPDATES = timedelta(seconds=12*60*60)  # 12h
+
+DOMAIN = 'gearbest'
+
+
+_ITEM_SCHEMA = vol.All(
+    vol.Schema({
+        vol.Exclusive(CONF_URL, 'XOR'): cv.string,
+        vol.Exclusive(CONF_ID, 'XOR'): cv.string,
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Optional(CONF_CURRENCY): cv.string
+    }), cv.has_at_least_one_key(CONF_URL, CONF_ID)
+)
+
+_ITEMS_SCHEMA = vol.Schema([_ITEM_SCHEMA])
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ITEMS): _ITEMS_SCHEMA,
+    vol.Required(CONF_CURRENCY): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Gearbest sensor."""
+    from gearbest_parser import CurrencyConverter
+    currency = config.get(CONF_CURRENCY)
+
+    sensors = []
+    items = config.get(CONF_ITEMS)
+
+    hass.data[DOMAIN] = CurrencyConverter()
+    hass.data[DOMAIN].update()
+
+    for item in items:
+        try:
+            sensor = GearbestSensor(hass, item, currency)
+            if sensor is not None:
+                sensors.append(sensor)
+        except ValueError as exc:
+            _LOGGER.error(exc)
+
+    def currency_update(event_time):
+        """Update currency list."""
+        hass.data[DOMAIN].update()
+
+    track_time_interval(hass,
+                        currency_update,
+                        MIN_TIME_BETWEEN_CURRENCY_UPDATES)
+
+    add_devices(sensors)
+
+
+class GearbestSensor(Entity):
+    """Implementation of the sensor."""
+
+    def __init__(self, hass, item, currency):
```

Comment:
Don't pass in `hass`. It will be set in the entity when it has been added to home assistant.
```py
def __init__(self, converter, item, currency):
```